miden-lib/src/wallets/mod.rs

```diff
@@ -17,10 +19,21 @@ use miden_objects::{
 /// Both methods require authentication. The authentication procedure is defined by the specified
 /// authentication scheme. Public key information for the scheme is stored in the account storage
 /// at slot 0.
+
 pub fn create_basic_wallet(
     init_seed: [u8; 32],
     auth_scheme: AuthScheme,
+    account_type: AccountType,
 ) -> Result<(Account, Word), AccountError> {
+    match account_type {
+        AccountType::FungibleFaucet | AccountType::NonFungibleFaucet => {
+            return Err(AccountError::AccountIdInvalidFieldElement(
+                "Basic wallet accounts cannot have a faucet account type".to_string(),
+            ))
+        }
+        _ => {}
+    }
```

Comment:
It might be cleaner to use the `matches!(account_type, AccountType::FungibleFaucet | AccountType::NonFungibleFaucet)` macro here.